app/controllers/admin/instances_controller.rb

```diff
@@ -66,7 +72,7 @@ def preload_delivery_failures!
     end
 
     def filtered_instances
-      InstanceFilter.new(limited_federation_mode? ? { allowed: true } : filter_params).results
+      InstanceFilter.new(limited_federation_mode? ? filter_params.merge(status: :allowed) : filter_params).results
```

Comment:
This allows usage of the `by_domain` and `availability` filters, whilst ensuring the status filter is always `allowed` in limited federation mode.